### PR TITLE
test: Makefile targets should cover all go modules

### LINKED ISSUE #337: Makefile targets should cover all go modules
In #322 we missed updating the `Makefile` so that the `test` target. We need `Makefile` and CI tests to cover all go modules.

### PR DESCRIPTION
**Pull request by cyberbono3**
Close #337

### CODE REVIEW

**Comment by cyberbono3** [comment]
If I run `make test`, the following output is being produced:
```
#go test -mod=readonly -tags='cgo ledger test_ledger_mock norace' ./...
find . -name go.mod -execdir go test ./... \;
?       github.com/regen-network/regen-ledger/types     [no test files]
ok      github.com/regen-network/regen-ledger/types/math        (cached)
?       github.com/regen-network/regen-ledger/types/module      [no test files]
?       github.com/regen-network/regen-ledger/types/module/client/cli   [no test files]
?       github.com/regen-network/regen-ledger/types/module/client/grpc_gateway  [no test files]
?       github.com/regen-network/regen-ledger/types/module/server       [no test files]
?       github.com/regen-network/regen-ledger/types/testutil    [no test files]
?       github.com/regen-network/regen-ledger/types/testutil/cli        [no test files]
?       github.com/regen-network/regen-ledger/types/testutil/network    [no test files]
ok      github.com/regen-network/regen-ledger/x/data    (cached)
?       github.com/regen-network/regen-ledger/x/data/client     [no test files]
?       github.com/regen-network/regen-ledger/x/data/client/testsuite   [no test files]
?       github.com/regen-network/regen-ledger/x/data/module     [no test files]
ok      github.com/regen-network/regen-ledger/x/data/server     (cached)
?       github.com/regen-network/regen-ledger/x/data/server/testsuite   [no test files]
ambiguous import: found package github.com/regen-network/regen-ledger/x/group in multiple modules:
        github.com/regen-network/regen-ledger/x/group (/home/ai/go/src/github.com/regen-network/regen-ledger/x/group)
        github.com/regen-network/regen-ledger v1.0.0 (/home/ai/go/pkg/mod/github.com/regen-network/regen-ledger@v1.0.0/x/group)
ambiguous import: found package github.com/regen-network/regen-ledger/x/group/client in multiple modules:
        github.com/regen-network/regen-ledger/x/group (/home/ai/go/src/github.com/regen-network/regen-ledger/x/group/client)
        github.com/regen-network/regen-ledger v1.0.0 (/home/ai/go/pkg/mod/github.com/regen-network/regen-ledger@v1.0.0/x/group/client)
ambiguous import: found package github.com/regen-network/regen-ledger/x/group/client/testsuite in multiple modules:
        github.com/regen-network/regen-ledger/x/group (/home/ai/go/src/github.com/regen-network/regen-ledger/x/group/client/testsuite)
        github.com/regen-network/regen-ledger v1.0.0 (/home/ai/go/pkg/mod/github.com/regen-network/regen-ledger@v1.0.0/x/group/client/testsuite)
ambiguous import: found package github.com/regen-network/regen-ledger/x/group/module in multiple modules:
        github.com/regen-network/regen-ledger/x/group (/home/ai/go/src/github.com/regen-network/regen-ledger/x/group/module)
        github.com/regen-network/regen-ledger v1.0.0 (/home/ai/go/pkg/mod/github.com/regen-network/regen-ledger@v1.0.0/x/group/module)
ambiguous import: found package github.com/regen-network/regen-ledger/x/group/server in multiple modules:
        github.com/regen-network/regen-ledger/x/group (/home/ai/go/src/github.com/regen-network/regen-ledger/x/group/server)
        github.com/regen-network/regen-ledger v1.0.0 (/home/ai/go/pkg/mod/github.com/regen-network/regen-ledger@v1.0.0/x/group/server)
ambiguous import: found package github.com/regen-network/regen-ledger/x/group/server/testsuite in multiple modules:
        github.com/regen-network/regen-ledger/x/group (/home/ai/go/src/github.com/regen-network/regen-ledger/x/group/server/testsuite)
        github.com/regen-network/regen-ledger v1.0.0 (/home/ai/go/pkg/mod/github.com/regen-network/regen-ledger@v1.0.0/x/group/server/testsuite)
simulation/operations.go:17:2: ambiguous import: found package github.com/regen-network/regen-ledger/types in multiple modules:
        github.com/regen-network/regen-ledger v1.0.0 (/home/ai/go/pkg/mod/github.com/regen-network/regen-ledger@v1.0.0/types)
        github.com/regen-network/regen-ledger/types v0.0.0-00010101000000-000000000000 (/home/ai/go/src/github.com/regen-network/regen-ledger/types)
ambiguous import: found package github.com/regen-network/regen-ledger/x/ecocredit in multiple modules:
        github.com/regen-network/regen-ledger/x/ecocredit (/home/ai/go/src/github.com/regen-network/regen-ledger/x/ecocredit)
        github.com/regen-network/regen-ledger v1.0.0 (/home/ai/go/pkg/mod/github.com/regen-network/regen-ledger@v1.0.0/x/ecocredit)
ambiguous import: found package github.com/regen-network/regen-ledger/x/ecocredit/client in multiple modules:
        github.com/regen-network/regen-ledger/x/ecocredit (/home/ai/go/src/github.com/regen-network/regen-ledger/x/ecocredit/client)
        github.com/regen-network/regen-ledger v1.0.0 (/home/ai/go/pkg/mod/github.com/regen-network/regen-ledger@v1.0.0/x/ecocredit/client)
ambiguous import: found package github.com/regen-network/regen-ledger/x/ecocredit/module in multiple modules:
        github.com/regen-network/regen-ledger/x/ecocredit (/home/ai/go/src/github.com/regen-network/regen-ledger/x/ecocredit/module)
        github.com/regen-network/regen-ledger v1.0.0 (/home/ai/go/pkg/mod/github.com/regen-network/regen-ledger@v1.0.0/x/ecocredit/module)
ambiguous import: found package github.com/regen-network/regen-ledger/x/ecocredit/server in multiple modules:
        github.com/regen-network/regen-ledger/x/ecocredit (/home/ai/go/src/github.com/regen-network/regen-ledger/x/ecocredit/server)
        github.com/regen-network/regen-ledger v1.0.0 (/home/ai/go/pkg/mod/github.com/regen-network/regen-ledger@v1.0.0/x/ecocredit/server)
ambiguous import: found package github.com/regen-network/regen-ledger/x/ecocredit/server/testsuite in multiple modules:
        github.com/regen-network/regen-ledger/x/ecocredit (/home/ai/go/src/github.com/regen-network/regen-ledger/x/ecocredit/server/testsuite)
        github.com/regen-network/regen-ledger v1.0.0 (/home/ai/go/pkg/mod/github.com/regen-network/regen-ledger@v1.0.0/x/ecocredit/server/testsuite)
ambiguous import: found package github.com/regen-network/regen-ledger/orm in multiple modules:
        github.com/regen-network/regen-ledger/orm (/home/ai/go/src/github.com/regen-network/regen-ledger/orm)
        github.com/regen-network/regen-ledger v1.0.0 (/home/ai/go/pkg/mod/github.com/regen-network/regen-ledger@v1.0.0/orm)
?       github.com/regen-network/regen-ledger/protoc-gen-go-cosmos2     [no test files]
ok      github.com/regen-network/regen-ledger/app       (cached)
?       github.com/regen-network/regen-ledger/app/regen [no test files]
Error: failed to parse log level (main:info,state:info,statesync:info,*:error): Unknown Level String: 'main:info,state:info,statesync:info,*:error', defaulting to NoLevel
Usage:
  regen init [moniker] [flags]

Flags:
      --chain-id string   genesis file chain-id, if left blank will be randomly created
  -h, --help              help for init
  -o, --overwrite         overwrite the genesis.json file
      --recover           provide seed phrase to recover existing key instead of creating

Global Flags:
      --home string         directory for config and data (default "/home/ai/.regen")
      --log_format string   The logging format (json|plain) (default "plain")
      --log_level string    The logging level (trace|debug|info|warn|error|fatal|panic) (default "info")
      --trace               print out full stack trace on errors

--- FAIL: TestInitCmd (0.01s)
    cmd_test.go:22: 
                Error Trace:    cmd_test.go:22
                Error:          Received unexpected error:
                                failed to parse log level (main:info,state:info,statesync:info,*:error): Unknown Level String: 'main:info,state:info,statesync:info,*:error', defaulting to NoLevel
                Test:           TestInitCmd
FAIL
FAIL    github.com/regen-network/regen-ledger/app/regen/cmd     0.056s
ok      github.com/regen-network/regen-ledger/store/lookup      (cached)
FAIL
```

I guess I have to fix ambiguous import issues here

File: Makefile [hunk 1]
```diff
@@ -253,9 +253,11 @@ $(TEST_TARGETS): run-tests
 
 run-tests:
 ifneq (,$(shell which tparse 2>/dev/null))
-	go test -mod=readonly -json $(ARGS) $(TEST_PACKAGES) | tparse
+	#go test -mod=readonly -json $(ARGS) $(TEST_PACKAGES) | tparse
+	find . -name go.mod -execdir go test ./... \;
```
Comment (robert-zaremba) [inline Makefile:257]: I think `-json` and `tparse` is missing here

**Review by aaronc — COMMENTED** [review]
This seems like a reasonable approach. What do you think @robert-zaremba ?

**Comment by aleem1314** [comment]
> I guess I have to fix ambiguous import issues here, right?

I think This is because of [regen-ledger](https://github.com/regen-network/regen-ledger/blob/master/orm/go.mod#L8) dependency. I think we can remove this.

**Comment by cyberbono3** [comment]
> > I guess I have to fix ambiguous import issues here, right?
> 
> I think this is because of [regen-ledger](https://github.com/regen-network/regen-ledger/blob/master/orm/go.mod#L8) dependency in the `orm` module. I think we can remove this.

Thanks, @aleem1314. Following your suggestion I have removed orm dep on regen ledger and it looks better now.
```
#go test -mod=readonly -tags='cgo ledger test_ledger_mock norace' ./...
find . -name go.mod -execdir go test ./... \;
?       github.com/regen-network/regen-ledger/types     [no test files]
ok      github.com/regen-network/regen-ledger/types/math        (cached)
?       github.com/regen-network/regen-ledger/types/module      [no test files]
?       github.com/regen-network/regen-ledger/types/module/client/cli   [no test files]
?       github.com/regen-network/regen-ledger/types/module/client/grpc_gateway  [no test files]
?       github.com/regen-network/regen-ledger/types/module/server       [no test files]
?       github.com/regen-network/regen-ledger/types/testutil    [no test files]
?       github.com/regen-network/regen-ledger/types/testutil/cli        [no test files]
?       github.com/regen-network/regen-ledger/types/testutil/network    [no test files]
ok      github.com/regen-network/regen-ledger/x/data    (cached)
?       github.com/regen-network/regen-ledger/x/data/client     [no test files]
?       github.com/regen-network/regen-ledger/x/data/client/testsuite   [no test files]
?       github.com/regen-network/regen-ledger/x/data/module     [no test files]
ok      github.com/regen-network/regen-ledger/x/data/server     (cached)
?       github.com/regen-network/regen-ledger/x/data/server/testsuite   [no test files]
--- FAIL: TestMsgCreateGroupAccount (0.00s)
    --- FAIL: TestMsgCreateGroupAccount/valid_admin_required (0.00s)
        msgs_test.go:208: 
                Error Trace:    msgs_test.go:208
                Error:          An error is expected but got nil.
                Test:           TestMsgCreateGroupAccount/valid_admin_required
--- FAIL: TestGroupAccountInfo (0.00s)
    --- FAIL: TestGroupAccountInfo/invalid_admin_account_address (0.00s)
        types_test.go:507: 
                Error Trace:    types_test.go:507
                Error:          An error is expected but got nil.
                Test:           TestGroupAccountInfo/invalid_admin_account_address
    --- FAIL: TestGroupAccountInfo/invalid_group_account_address (0.00s)
        types_test.go:507: 
                Error Trace:    types_test.go:507
                Error:          An error is expected but got nil.
                Test:           TestGroupAccountInfo/invalid_group_account_address
FAIL
FAIL    github.com/regen-network/regen-ledger/x/group   0.221s
?       github.com/regen-network/regen-ledger/x/group/client    [no test files]
?       github.com/regen-network/regen-ledger/x/group/client/testsuite  [no test files]
?       github.com/regen-network/regen-ledger/x/group/exported  [no test files]
?       github.com/regen-network/regen-ledger/x/group/module    [no test files]
--- FAIL: TestServer (0.03s)
    --- FAIL: TestServer/TestExecProposal (0.01s)
        --- FAIL: TestServer/TestExecProposal/executable_when_failed_before (0.00s)
            suite.go:1933: 
                        Error Trace:    suite.go:1933
                                                                suite.go:77
                        Error:          Not equal: 
                                        expected: "EXECUTOR_RESULT_SUCCESS"
                                        actual  : "EXECUTOR_RESULT_FAILURE"
                                    
                                        Diff:
                                        --- Expected
                                        +++ Actual
                                        @@ -1 +1 @@
                                        -EXECUTOR_RESULT_SUCCESS
                                        +EXECUTOR_RESULT_FAILURE
                        Test:           TestServer/TestExecProposal/executable_when_failed_before
FAIL
FAIL    github.com/regen-network/regen-ledger/x/group/server    0.081s
?       github.com/regen-network/regen-ledger/x/group/server/testsuite  [no test files]
?       github.com/regen-network/regen-ledger/x/group/simulation        [no test files]
FAIL
?       github.com/regen-network/regen-ledger/x/ecocredit       [no test files]
?       github.com/regen-network/regen-ledger/x/ecocredit/client        [no test files]
?       github.com/regen-network/regen-ledger/x/ecocredit/module        [no test files]
ok      github.com/regen-network/regen-ledger/x/ecocredit/server        0.050s
?       github.com/regen-network/regen-ledger/x/ecocredit/server/testsuite      [no test files]
?       github.com/regen-network/regen-ledger/x/ecocredit/util  [no test files]
ok      github.com/regen-network/regen-ledger/orm       0.052s
?       github.com/regen-network/regen-ledger/protoc-gen-go-cosmos2     [no test files]
ok      github.com/regen-network/regen-ledger/app       (cached)
?       github.com/regen-network/regen-ledger/app/regen [no test files]
Error: failed to parse log level (main:info,state:info,statesync:info,*:error): Unknown Level String: 'main:info,state:info,statesync:info,*:error', defaulting to NoLevel
Usage:
  regen init [moniker] [flags]

Flags:
      --chain-id string   genesis file chain-id, if left blank will be randomly created
  -h, --help              help for init
  -o, --overwrite         overwrite the genesis.json file
      --recover           provide seed phrase to recover existing key instead of creating

Global Flags:
      --home string         directory for config and data (default "/home/ai/.regen")
      --log_format string   The logging format (json|plain) (default "plain")
      --log_level string    The logging level (trace|debug|info|warn|error|fatal|panic) (default "info")
      --trace               print out full stack trace on errors

--- FAIL: TestInitCmd (0.01s)
    cmd_test.go:22: 
                Error Trace:    cmd_test.go:22
                Error:          Received unexpected error:
                                failed to parse log level (main:info,state:info,statesync:info,*:error): Unknown Level String: 'main:info,state:info,statesync:info,*:error', defaulting to NoLevel
                Test:           TestInitCmd
FAIL
FAIL    github.com/regen-network/regen-ledger/app/regen/cmd     0.070s
ok      github.com/regen-network/regen-ledger/store/lookup      (cached)
FAIL
```

**Comment by blushi** [comment]
Why does "Tests / Unit tests" is marked as  successful while it's actually failing: https://github.com/regen-network/regen-ledger/pull/357/checks?check_run_id=2658718646 ?
I guess those tests were silently failing since 0.43 upgrade PR https://github.com/regen-network/regen-ledger/pull/349.

**Comment by aleem1314** [comment]
>  Also it seems like all tests are run even without `EXPERIMENTAL=true`.

Yes, We have to add the `// +build experimental` flag to all test files.

**Comment by blushi** [comment]
@cyberbono3 could you fix the go.sum updates issue?

**Review by blushi — APPROVED** [review]

**Review by aleem1314 — APPROVED** [review]

**Comment by aleem1314** [comment]
[codecov](https://app.codecov.io/gh/regen-network/regen-ledger/compare/357/tree) too not covering all submodules. Let's create an issue for this.

**Review by likhita-809 — APPROVED** [review]

**Comment by blushi** [comment]
> [codecov](https://app.codecov.io/gh/regen-network/regen-ledger/compare/357/tree) too not covering all submodules. Let's create an issue for this.

done https://github.com/regen-network/regen-ledger/issues/368

**Pull request merged**